migrations/2_deploy_registry.js

```diff
@@ -0,0 +1,52 @@
+/* globals artifacts web3 */
+const ENS = require('ethereum-ens');
+const { deployUpgradeableContract } = require('../test/helpers/contracts');
+const getNamedAccounts = require('../test/helpers/getNamedAccounts');
+
+const ContractRegistryV0_1_0 = artifacts.require('ContractRegistryV0_1_0');
+
+module.exports = (deployer, network, accounts) => {
+  deployer.then(async () => {
+    let registry, adminAccountAddress;
+    if (network === 'ropsten') {
+      adminAccountAddress = accounts[0];
```

Comment:
We should force a sanity check here so that we can't deploy from a truffle account

Reply:
I'm trying to think of how to do that. It seems like the truffle team periodically changes the default mnemonic that gets used. I will think on this some and try to come up with something good.

Reply:
I'm sure the seed exists in one of the truffle repos. Maybe query it for changes

Reply:
You might also check here for some example? Although I think our upgradeable implementation is superior to any I've seen now 😏 https://github.com/JoinColony/colonyNetwork/tree/develop/helpers

Reply:
The open zeppelin example you linked to does not use a shared mnemonic. Looks like it works exactly the same as this with a mnemonic specified via an environment variable.

Reply:
I get the idea. I just don't see a documented way of specifying a mnemonic for truffle to use short of instantiating your own provider, and even when you do that, `truffle develop` ignores the provider you specified.

In any case, I'm not that concerned about someone accidentally deploying to main net using a public seed phrase so long as the ENS entry always points to the right place. And once the registry is properly deployed to main net with the multisig wallet as the owner, it's not like you'll be able to change anything using truffle keys anyway.

Reply:
I am pretty worried actually, but not for the reason described here. Mainly for one in which is using another edcase that addes account[0] to multisig, transfers ownership or sets owner to. 

BUT, there's actually a pretty straightforward solution I think we both overlooked. *Whitelist our two developer addresses. *